src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java

```diff
@@ -157,6 +157,7 @@ public static Builder builder(String name, SchedulerPoolsConfig config) {
   protected final ThreadGroup customWaitGroup;
   protected final ThreadGroup customCallerRunsGroup;
   protected final ThreadGroup customCallerRunsAnsWaitGroup;
+  protected Set<ThreadGroup> waitGroups;
```

Comment:
move this field to the subclasses where this is set, add an abstract getter to use in this class

Reply:
Done.